src/main/java/mate/academy/service/impl/ShoppingCartServiceImpl.java

```diff
@@ -0,0 +1,53 @@
+package mate.academy.service.impl;
+
+import java.util.Collections;
+import java.util.List;
+import mate.academy.dao.ShoppingCartDao;
+import mate.academy.dao.TicketDao;
+import mate.academy.lib.Inject;
+import mate.academy.lib.Service;
+import mate.academy.model.MovieSession;
+import mate.academy.model.ShoppingCart;
+import mate.academy.model.Ticket;
+import mate.academy.model.User;
+import mate.academy.service.ShoppingCartService;
+
+@Service
+public class ShoppingCartServiceImpl implements ShoppingCartService {
+    @Inject
+    private ShoppingCartDao shoppingCartDao;
+    @Inject
+    private TicketDao ticketDao;
+
+    @Override
+    public void addSession(MovieSession movieSession, User user) {
+        ShoppingCart cart = shoppingCartDao.getByUser(user).orElseThrow(() ->
+                new RuntimeException("Optional user by id: " + user.getId()
+                        + " is empty"));
```

Comment:
```suggestion
        ShoppingCart cart = getByUser(user);
```
this line will throw an exception if cart isn't found)